src/triangle.rs

```diff
@@ -151,12 +160,18 @@ impl Triangle {
         let centroid = self.centroid().to_vec2();
 
         [
-            (centroid - self.a.to_vec2()).to_size(),
-            (centroid - self.b.to_vec2()).to_size(),
-            (centroid - self.c.to_vec2()).to_size(),
+            (self.a.to_vec2() - centroid).to_size(),
+            (self.b.to_vec2() - centroid).to_size(),
+            (self.c.to_vec2() - centroid).to_size(),
         ]
     }
 
+    /// The perimeter of the [`Triangle`]
+    #[inline]
+    fn perimeter(&self) -> f64 {
+        self.a.distance(self.b) + self.b.distance(self.c) + self.c.distance(self.a)
+    }
+
```

Comment:
we have general `perimeter` and `area` functions but we also have specific ones for right angled triangles. should we add more specific ones for equilateral triangles and stuff or is it too much clutter?

also, i dont see the need to implement perimeter and area here when its in the `Shape` trait? this is also done for `Rect`.